rpcs3/Emu/CPU/CPUThread.cpp

```diff
@@ -767,6 +776,30 @@ bool cpu_thread::check_state() noexcept
 			return store;
 		}).first;
 
+		if (state0 & cpu_flag::preempt)
+		{
+			if (cpu_flag::wait - state0)
+			{
+				// Yield itself
+				state.wait(state1, atomic_wait_timeout{20'000});
+			}
+
+			if (const u128 bits = s_cpu_bits)
+			{
+				reader_lock lock(s_cpu_lock);
+
+				cpu_counter::for_all_cpu(bits & s_cpu_bits, [](cpu_thread* cpu)
```

Comment:
Does this ignore rsx::thread? I mean, it is a cpu_thread, no?

Reply:
Yes but it's not registered by cpu_counter::add for other reasons.

Reply:
Got it.